flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -87,6 +89,7 @@ public void endInput() throws IOException {
     // remaining completed files to downstream before closing the writer so that we won't miss any
     // of them.
     flush();
+    ended = true;
```

Comment:
nit: I think we probably don't need this boolean flag. Instead, we can mark writer as null after `writer.complete()`. then we can just use the null check on writer inside the `flush()` method.

Reply:
Which `writer.complete()` call are you suggesting? It seems to me that `writer.complete()` is called in `flush`, which is explicitly called in multiple places.

Reply:
> nit: I think we probably don't need this boolean flag. Instead, we can mark writer as null after `writer.complete()`. then we can just use the null check on writer inside the `flush()` method.

Can you add an isclosed method to write to determine whether it has been closed

Reply:
Oh never mind I see: we’re calling create and instantiating a new writer after the other places that call `flush`, but not this one. Arguably we don’t need the variable then.

But we should consider cleaning up the above comment and adding information about the `null` check being used as the guard here or where the check happens.

Reply:
I understand. I'll go and fix it

Reply:
I agree that a method might be helpful for readability.

`isClosed` seems odd to me personally as there is a `close` method on the class and we haven’t necessarily called it yet. Do you suggest adding `isClosed` to the writer itself?

Reply:
If the writer class isn’t being touched (I haven’t looked that closely yet) or if the method will be placed on this class, I’d consider naming the method something like `shouldFlushWriter` or something.

But it does seem that checking for null is technically sufficient overall. Thanks @xuzhiwen1255.

Reply:
Also, if it’s possible to add a test for this, that would be great! However, the test seems admittedly flaky by nature so if it’s not consistently possible don’t worry too much about it. But looking into it would be great!

Reply:
It's me. I think null is enough

Reply:
The nice thing about adding a new method is that the code will look a little bit cleaner, and if we're adding a new method, we'll name it something that matches
@kbendick

Reply:
I'll try to add a test

Reply:
+1 for adding a unit test. I think it can be deterministic. take a look at the `TestIcebergStreamWriter`. we can trigger the `endInput` first, then the `prepareSnapshotPreBarrier` to simulate this scenario.

Reply:
> +1 for adding a unit test. I think it can be deterministic. take a look at the `TestIcebergStreamWriter`. we can trigger the `endInput` first, then the `prepareSnapshotPreBarrier` to simulate this scenario.

I added a test and passed it.